jerry-core/ecma/base/ecma-helpers-number.c

```diff
@@ -718,6 +718,32 @@ ecma_number_multiply (ecma_number_t left_num, /**< left operand */
   return left_num * right_num;
 } /* ecma_number_multiply */
 
+/**
+ * ECMA-integer number multiplication.
+ *
+ * @return number - result of multiplication.
+ */
+inline ecma_value_t __attr_always_inline___
+ecma_integer_multiply (ecma_integer_value_t left_integer, /**< left operand */
+                       ecma_integer_value_t right_integer) /**< right operand */
+{
+#ifdef __GNUC__
+  /* Check if left_integer is power of 2 */
+  if (unlikely (__builtin_popcount ((unsigned int) left_integer) == 1))
```

Comment:
Power of 2 has an easier check (and compiler independent):
```
if ((a & (a - 1)) == 0)
  a_is_power_of_2
```
The popcount is only cheap on 64 bit sparc systems. Not exactly our target.

Reply:
Good point!